src/hash_generator/dlNormalizer.js

```diff
@@ -0,0 +1,398 @@
+/**
+ * (c) Copyright Reserved EVRYTHNG Limited 2021. All rights reserved.
+ * Use of this material is subject to license.
+ * Copying and unauthorised use of this material strictly prohibited.
+ */
+
+/* eslint-disable no-useless-escape */
+
+/**
+ * it protects some character with '%..'
+ * @param {string} input - the string to protect
+ * @return {string} the updated string
+ */
+export const webUriPercentEncoder = (input) => input.split('!').join('%21')
```

Comment:
Would something like `encodeURIComponent` be safer here than an explicit list of cases? Unless this list is exhaustive.

Reply:
I took this list from the original algorithm. Using `encodeURIComponent` won't give the same output for the same input..